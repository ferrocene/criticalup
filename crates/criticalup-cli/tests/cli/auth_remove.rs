// SPDX-FileCopyrightText: The Ferrocene Developers
// SPDX-License-Identifier: MIT OR Apache-2.0

use crate::assert_output;
use crate::utils::{TestEnvironment, MOCK_AUTH_TOKENS};
use serde::Deserialize;

#[tokio::test]
async fn help_message() {
    let test_env = TestEnvironment::prepare().await;
    assert_output!(test_env.cmd().args(["auth", "remove", "--help"]));
}

#[tokio::test]
async fn token_missing() {
    let test_env = TestEnvironment::prepare().await;

    assert_output!(test_env.cmd().args(["auth", "remove"]));
    assert_eq!(0, test_env.requests_served_by_mock_download_server().await);

    // Ensure no state file was created by just running remove.
    assert!(!test_env.root().join("state.json").exists());
}

#[tokio::test]
async fn token_present() {
    #[derive(Deserialize)]
    struct State {
        authentication_token: Option<String>,
    }

    let test_env = TestEnvironment::prepare().await;
    assert!(test_env
        .cmd()
        .args(["auth", "set", MOCK_AUTH_TOKENS[0].0])
        .output()
        .await
        .expect("failed to set token")
        .status
        .success());

    assert_output!(test_env.cmd().args(["auth", "remove"]));
    assert_eq!(1, test_env.requests_served_by_mock_download_server().await);

    let state: State =
        serde_json::from_slice(&std::fs::read(test_env.root().join("state.json")).unwrap())
            .unwrap();
    assert!(state.authentication_token.is_none());
}
