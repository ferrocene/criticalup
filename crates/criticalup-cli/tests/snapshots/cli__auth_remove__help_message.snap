---
source: crates/criticalup-cli/tests/cli/auth_remove.rs
expression: repr
---
exit: exit status: 0

empty stdout

stderr
------
Remove the authentication token used to interact with the download server

Usage:
  criticalup-test auth remove [OPTIONS]

Options:
  -v, --verbose...                  Enable debug logs, -vv for trace
      --log-format <LOG_FORMAT>     Which logger to use [default: default] [possible values: default, pretty, tree, json]
      --log-level [<LOG_LEVEL>...]  Tracing directives
  -h, --help                        Print help
------
